rewrite-core/src/main/java/org/openrewrite/marker/GitProvenance.java

```diff
@@ -85,54 +107,52 @@ public class GitProvenance implements Marker {
         if (baseUrl.startsWith("git@")) {
             baseUrl = baseUrl.substring(4);
         }
-        String remainder = origin.substring(origin.indexOf(baseUrl) + baseUrl.length());
-        if (remainder.startsWith(":")) {
-            remainder = remainder.substring(1);
-        }
-        if (remainder.startsWith("/")) {
-            remainder = remainder.substring(1);
-        }
+        String remainder = origin.substring(origin.indexOf(baseUrl) + baseUrl.length())
+                .replaceFirst("^:", "")
+                .replaceFirst("^/", "");
         return remainder.substring(0, remainder.lastIndexOf('/'));
     }
 
-    /**
-     * There is too much variability in how different git hosting services arrange their organizations to reliably
-     * determine the organization component of the URL without additional information. The version of this method
-     * which accepts a "baseUrl" parameter should be used instead
-     */
-    @Deprecated
     public @Nullable String getOrganizationName() {
-        if (origin == null) {
+        if (getGitRemote() == null) {
             return null;
         }
-        try {
-            String path = new URIish(origin).getPath();
-            // Strip off any trailing repository name
-            path = path.substring(0, path.lastIndexOf('/'));
-            // Strip off any leading sub organization names
-            return path.substring(path.lastIndexOf('/') + 1);
-        } catch (URISyntaxException e) {
+        return getGitRemote().getOrganization();
+    }
+
+    public @Nullable String getRepositoryName() {
+        if (getGitRemote() == null) {
             return null;
         }
+        return getGitRemote().getRepositoryName();
     }
 
-    public @Nullable String getRepositoryName() {
-        if (origin == null) {
+    public @Nullable String getRepositoryPath() {
```

Comment:
I'd prefer we didn't change this name. I think you are probably trying to be more descriptive for the GitLab scenario, but I don't think that rises to the level of making a breaking change here.

Reply:
The name is not changed, this is a new method (used to retrieve the path, or organization + "/" + reponame)